### ISSUE TITLE
'WRF+ROMS+SWAN' issue cowast

### ISSUE THREAD

**Issue by jiangdongZ** [jcwarner-usgs/COAWST#195]
I have successfully run the sandy calculation，The following problem occurred when I changed a region for simulation.
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 200898 RUNNING AT ln01
=   EXIT CODE: 135
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Bus error (signal 7)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions
[1.txt](https://github.com/DOI-USGS/COAWST/files/13459346/1.txt)
[coupling_sandy.txt](https://github.com/DOI-USGS/COAWST/files/13459375/coupling_sandy.txt)
[namelist.txt](https://github.com/DOI-USGS/COAWST/files/13459376/namelist.txt)
[ocean_sandy.txt](https://github.com/DOI-USGS/COAWST/files/13459377/ocean_sandy.txt)
[sandy.txt](https://github.com/DOI-USGS/COAWST/files/13459379/sandy.txt)
[swan_sandy.txt](https://github.com/DOI-USGS/COAWST/files/13459380/swan_sandy.txt)
[swan_sandy_ref3.txt](https://github.com/DOI-USGS/COAWST/files/13459384/swan_sandy_ref3.txt)
I need help, thank you very much. Or can you give me an example from other sea areas?

**Comment by jcwarner-usgs** [comment]
not sure .
looks like all the modes started, exchanged fields, and then after 14 time steps something went wrong.
did you get any Err* files (this would be swan issues).
can you look at the wrfout and roms his files?  
are there any other files that can help determine what went wrong?

**Comment by jiangdongZ** [comment]
Yes, both my WRF and ROMS have file outputs. Only SWAN doesn't have these; it's my SWAN output files.
[Errfile01.txt](https://github.com/DOI-USGS/COAWST/files/13522177/Errfile01.txt)
[PRINT01.txt](https://github.com/DOI-USGS/COAWST/files/13522178/PRINT01.txt)
[PRINT02.txt](https://github.com/DOI-USGS/COAWST/files/13522180/PRINT02.txt)

**Comment by jiangdongZ** [comment]
Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0 0x2B443989D6D7
https://github.com/DOI-USGS/COAWST/issues/1 0x2B443989DD1E
https://github.com/DOI-USGS/COAWST/issues/2 0x2B443A9E63FF
#0 0x2B621385A6D7
https://github.com/DOI-USGS/COAWST/issues/1 0x2B621385AD1E
https://github.com/DOI-USGS/COAWST/issues/2 0x2B62149A33FF
https://github.com/DOI-USGS/COAWST/issues/3 0x96EAAC in swoexd_
https://github.com/DOI-USGS/COAWST/issues/3 0x96EAAC in swoexd_
https://github.com/DOI-USGS/COAWST/issues/4 0x985D35 in swouta_
https://github.com/DOI-USGS/COAWST/issues/4 0x985D35 in swouta_
https://github.com/DOI-USGS/COAWST/issues/5 0x8CC2E8 in swan_cpl_
https://github.com/DOI-USGS/COAWST/issues/5 0x8CC2E8 in swan_cpl_
https://github.com/DOI-USGS/COAWST/issues/6 0x85AA9D in waves_control_mod_MOD_swan_driver_run
https://github.com/DOI-USGS/COAWST/issues/6 0x85AA9D in waves_control_mod_MOD_swan_driver_run
https://github.com/DOI-USGS/COAWST/issues/7 0x40DCDC in MAIN at master.f90:?
https://github.com/DOI-USGS/COAWST/issues/7 0x40DCDC in MAIN at master.f90:?
I found it is due to the SWAN model."I found that it runs when I provide the SWAN wind field file, but it cannot run when I don't provide it (it stops after running for 30 minutes). In the coupling, shouldn't I leave it blank?"
[scs_swan_cowast.txt](https://github.com/DOI-USGS/COAWST/files/13523771/scs_swan_cowast.txt)

**Comment by jcwarner-usgs** [comment]
you can follow the Sandy example. and some of this information is in the COAWST manual.

if you have all 3 coupled (roms swan wrf), then in the swan input file you need:


&& KEYWORD for number of nested SWAN grids.
NSGRIDS 2   (<----   list the number of  swan grids)

&& KEYWORDS TO CREATE AND READ COMPUTATIONAL GRID &&
CGRID CURVILINEAR 86 64 EXC 9.999000e+003 9.999000e+003 CIRCLE 36 0.04 1.0 24
READGRID COORDINATES 1 'Projects/Sandy/Sandy_swan_coord.grd' 4 0 0 FREE 

&& KEYWORDS TO CREATE AND READ BATHYMETRY GRID &&
INPGRID BOTTOM CURVILINEAR 0 0 86 64 EXC 9.999000e+003
READINP BOTTOM 1 'Projects/Sandy/Sandy_swan_bathy.bot' 4 0 FREE 

&& KEYWORD TO CREATE CURRENT GRID &&    (<----   list these commands, it tells swan to allow currents to affect waves) 
INPGRID CURRENT CURVILINEAR 0 0 86 64 EXC 9.999000e+003 &
       NONSTATIONARY 20121028.120000 2 DAY 20121030.120000
 
&& KEYWORD TO CREATE WATER LEVEL GRID &&   (<----   list these commands, it tells swan to allow varying water levels) 
INPGRID WLEV CURVILINEAR 0 0 86 64 EXC 9.999000e+003 &
       NONSTATIONARY 20121028.120000 2 DAY 20121030.120000

&& KEYWORD TO CREATE BOTTOM FRIC GRID &&     (<----   list these commands, it tells swan to allow vary bottom fric) 
INPGRID FRIC CURVILINEAR 0 0 86 64 EXC 9.999000e+003 &
       NONSTATIONARY 20121028.120000 2 DAY 20121030.120000

&& KEYWORD TO CREATE WIND GRID &&       (<----   list these commands, it tells swan that wind will be on a grid) 
( this wind grid could just be the size of wrf, or the size of the swan grid, it does not matter.  but you need to list a 
grid here so swan knows that the winds are coming in)
INPGRID WIND REGULAR -105 10 0 220 160 0.25 0.25 &
       NONSTATIONARY 20121028.000000 6 HR 20121031.000000

&READINP WIND 1 'Projects/Sandy/swan_GFS_Oct2012.dat' 4 0 FREE         (<----  here we comment the wind read line because it will come through the coupler)
